src/wazuh_modules/vulnerability_detector/wm_vuln_detector_evr.c

```diff
@@ -432,7 +432,37 @@ static int nvd_verrevcmp(const char *a, const char *b, int revision)
         result = deb_verrevcmp(rev, b, 1);
 
         os_free(rev);
+
     } else {
+        // When comparing versions, we are comparing the number of characters
+        // of the NVD version contains for certains cases only
+
+        size_t a_size = strlen(a);
+        size_t b_size = strlen(b);
+
+        if (a_size > b_size) {
+            int i;
+            for (i = 0; i < (int)a_size - 1; i++) {
+                if ((a[i] == '.'
+                    || a[i] == '~'
+                    || a[i] == '+')
+                    && !c_isdigit(a[i + 1])) {
```

Comment:
What if there is a number after a `~` or a `+`?

Reply:
Included that condition https://github.com/wazuh/wazuh/pull/5362/commits/0362989ab8f79f3cc0be939e9ba5fb5801f8cee1